fluss-client/src/main/java/com/alibaba/fluss/client/utils/ConverterUtils.java

```diff
@@ -0,0 +1,610 @@
+ *
+ * // Convert a POJO to GenericRow
+ * Order order = new Order(1001L, 5001L, 10, "123 Athens");
+ * GenericRow row = converter.toRow(order);
+ *
+ * // Convert a GenericRow back to POJO
+ * Order convertedOrder = converter.fromRow(row);
+ * }</pre>
+ *
+ * <p>Note: Nested POJO fields are not supported in the current implementation.
+ *
+ * @param <T> The POJO type to convert
+ */
+public class ConverterUtils<T> {
+    private static final Logger LOG = LoggerFactory.getLogger(ConverterUtils.class);
+
+    /** Cache for converters to avoid repeated reflection operations. */
+    private static final ConcurrentHashMap<CacheKey, ConverterUtils<?>> CONVERTER_CACHE =
+            MapUtils.newConcurrentHashMap();
+
+    /** Map of supported Java types for each DataTypeRoot. */
+    private static final Map<DataTypeRoot, Set<Class<?>>> SUPPORTED_TYPES = new HashMap<>();
+
+    static {
+        SUPPORTED_TYPES.put(DataTypeRoot.BOOLEAN, orderedSet(Boolean.class, boolean.class));
+        SUPPORTED_TYPES.put(DataTypeRoot.TINYINT, orderedSet(Byte.class, byte.class));
+        SUPPORTED_TYPES.put(DataTypeRoot.SMALLINT, orderedSet(Short.class, short.class));
+        SUPPORTED_TYPES.put(DataTypeRoot.INTEGER, orderedSet(Integer.class, int.class));
+        SUPPORTED_TYPES.put(DataTypeRoot.BIGINT, orderedSet(Long.class, long.class));
+        SUPPORTED_TYPES.put(DataTypeRoot.FLOAT, orderedSet(Float.class, float.class));
+        SUPPORTED_TYPES.put(DataTypeRoot.DOUBLE, orderedSet(Double.class, double.class));
+        SUPPORTED_TYPES.put(
+                DataTypeRoot.CHAR, orderedSet(String.class, Character.class, char.class));
+        SUPPORTED_TYPES.put(
+                DataTypeRoot.STRING, orderedSet(String.class, Character.class, char.class));
+        SUPPORTED_TYPES.put(DataTypeRoot.BINARY, orderedSet(byte[].class));
+        SUPPORTED_TYPES.put(DataTypeRoot.BYTES, orderedSet(byte[].class));
+        SUPPORTED_TYPES.put(DataTypeRoot.DECIMAL, orderedSet(BigDecimal.class));
+        SUPPORTED_TYPES.put(DataTypeRoot.DATE, orderedSet(LocalDate.class));
+        SUPPORTED_TYPES.put(DataTypeRoot.TIME_WITHOUT_TIME_ZONE, orderedSet(LocalTime.class));
+        SUPPORTED_TYPES.put(
+                DataTypeRoot.TIMESTAMP_WITHOUT_TIME_ZONE, orderedSet(LocalDateTime.class));
+        SUPPORTED_TYPES.put(
+                DataTypeRoot.TIMESTAMP_WITH_LOCAL_TIME_ZONE,
+                orderedSet(Instant.class, OffsetDateTime.class));
+
+        // TODO: Add more types when https://github.com/apache/fluss/issues/816 is merged
+    }
+
+    /** Interface for field conversion from POJO field to Fluss InternalRow field. */
+    private interface FieldToRowConverter {
+        Object convert(Object obj) throws IllegalAccessException;
+    }
+
+    /** Interface for field conversion from Fluss InternalRow field to POJO field. */
+    private interface RowToFieldConverter {
+        Object convert(InternalRow row, int pos) throws IllegalAccessException;
```

Comment:
`throws Exception` throws any exception to make the wrapper catch all the exception.